quixstreams/sinks/community/file/sink.py

```diff
@@ -0,0 +1,167 @@
+import logging
+import re
+from pathlib import Path
+from typing import Literal, Union
+
+from quixstreams.sinks import BatchingSink, SinkBatch
+
+from .formats import Format, JSONFormat, ParquetFormat
+
+__all__ = ["FileSink", "InvalidFormatError"]
+
+logger = logging.getLogger(__name__)
+
+FormatName = Literal["json", "parquet"]
+
+_FORMATS: dict[FormatName, Format] = {
+    "json": JSONFormat(),
+    "parquet": ParquetFormat(),
+}
+
+_UNSAFE_CHARACTERS_REGEX = re.compile(r"[^a-zA-Z0-9 ._]")
+
+
+class InvalidFormatError(Exception):
+    """
+    Raised when the format is specified incorrectly.
+    """
+
+
+class FileSink(BatchingSink):
+    """
+    Writes batches of data to files on disk using specified formats.
+
+    Messages are grouped by their topic and partition. Data from messages with
+    the same topic and partition are saved in the same directory. Each batch of
+    messages is serialized and saved to a file within that directory. Files are
+    named using the batch's starting offset to ensure uniqueness and order.
+
+    If `append` is set to `True`, the sink will attempt to append data to an
+    existing file rather than creating a new one. This is only supported for
+    formats that allow appending.
+    """
+
+    def __init__(
+        self, output_dir: str, format: Union[FormatName, Format], append: bool = False
+    ) -> None:
+        """
+        Initializes the FileSink.
+
+        :param output_dir: The directory where files will be written.
+        :param format: The data serialization format to use. This can be either a
+            format name ("json", "parquet") or an instance of a `Format`
+            subclass.
+        :param append: If `True`, data will be appended to existing files when possible.
+            Note that not all formats support appending. Defaults to `False`.
+        :raises ValueError: If `append` is `True` but the specified format does not
+            support appending.
+        """
+        super().__init__()
+        self._format = self._resolve_format(format)
+        self._output_dir = output_dir  # TODO: validate
+        if append and not self._format.supports_append:
+            raise ValueError(f"`{format}` format does not support appending.")
+        self._append = append
+        self._file_mode = "ab" if append else "wb"
+        logger.info(f"Files will be written to '{self._output_dir}'.")
+
+    def write(self, batch: SinkBatch) -> None:
+        """
+        Writes a batch of data to files on disk, grouping data by topic and partition.
+
+        If `append` is `True` and an existing file is found, data will be appended to
+        the last file. Otherwise, a new file is created based on the batch's starting
+        offset.
+
+        :param batch: The batch of data to write.
+        """
+
+        # Generate directory based on topic and partition
+        # Generate filename based on a padded offset
+        # Returns existing file path if append is True
```

Comment:
Looks like the comment refers to something different here

Reply:
Yeah the wording is poor here. I will skip this comment since there is a big on in the method itself.